org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/TaskViewState.java

```diff
@@ -54,6 +56,7 @@ public void load() {
         this.linkToSelection = prefs.getBoolean(PREF_LINK_TO_SELECTION, false);
         this.headerNameColumnWidth = prefs.getInt(PREF_HEADER_NAME_COLUMN_WIDTH, 200);
         this.headerDescriptionColumnWidth = prefs.getInt(PREF_HEADER_DESCRIPTION_COLUMN_WIDTH, 400);
+        this.groupProjects = prefs.getBoolean(PREF_GROUP_PROJECTS_BY_HIERARCHY, true);
```

Comment:
Please add test coverage to `TaskViewContentTest` for the new hierarchy.

The Oomph buildship setup should give you a working IDE setup. Just a couple of notes:
- sometimes you have to activate the `tooling-e47.target` target definition file to compile everything.
- the UI tests are launched on the UI thread by default which causes the Buildship UI tests to hang. You have to disable it in the run configurations to make them work.

Reply:
Thanks for this tips, I will add this tests in recent days